api/v1beta1/machineset_types.go

```diff
@@ -129,6 +134,27 @@ const (
 	OldestMachineSetDeletePolicy MachineSetDeletePolicy = "Oldest"
 )
 
+// ANCHOR: RemediationStrategy
+
+// RemediationStrategy allows to define how control plane machine remediation happens.
+type RemediationStrategy struct {
+	// MaxInFlight determines how many in flight remediations should the MachineSet issue.
+	//
+	// When MachineHealthChecks marks a set of Machines as unhealthy, the default behavior
+	// is to delete all of those at once, and start replacing them immediately.
+	//
+	// To have control over how many Machines should be remediated, the MaxInFlight can be set
+	// to a fixed number or a percentage.
+	// Example: when this is set to 20%, the MachineSet controller deletes at most 20% of
+	// the machines under management at any given point in time.
+	//
+	// If not set, remedation has unbounded limit.
+	// +optional
+	MaxInFlight *intstr.IntOrString `json:"maxInFlight,omitempty"`
```

Comment:
Should this be

```suggestion
	// maxParallelism the maximum number of machines that can be remediating at a given time.
	// ...
	maxParallelism *intstr.IntOrString `json:"maxParallelism,omitempty"`
```
(prior art: spec.parallelism in Kubernetes's Jobs; another alternative we can cosider is to use concurrency)

Note: given that when we use MD I consider MS as an implementation detail the users should not care about, I think the should consider all the machines under the MD and not be propagated down to MS

Reply:
We can change it although was trying not to pick anything that resembles concurrency or parallelism given that the remediation is mostly a set and continue (delete) operation in this context, we just limit the amount of delete operation that are occurring at the same time, in flight is somewhat in line and reads well in `spec.remediationStrategy.maxInFlight`

Reply:
fwiw I am leaning towards maxInFlight as well.